src/microbe_stage/systems/ProcessSystem.cs

```diff
@@ -227,18 +228,23 @@ public static EnergyBalanceInfo ComputeEnergyBalance(IReadOnlyList<OrganelleTemp
     ///   If true, then the required input compounds to run at the given energy balance are stored per energy producer
     /// </param>
     /// <param name="cache">Auto-Evo Cache for speeding up the function</param>
-    public static EnergyBalanceInfo ComputeEnergyBalance(IReadOnlyList<OrganelleTemplate> organelles,
+    /// <param name="result">
+    ///   The resulting energy balance. If not null, the results are added to the passed value.
+    /// </param>
+    public static void ComputeEnergyBalance(IReadOnlyList<OrganelleTemplate> organelles,
         IBiomeConditions biome, MembraneType membrane, Vector3 onlyMovementInDirection,
         bool includeMovementCost, bool isPlayerSpecies, WorldGenerationSettings worldSettings,
-        CompoundAmountType amountType, bool calculateRequiredResources, SimulationCache? cache)
+        CompoundAmountType amountType, bool calculateRequiredResources, SimulationCache? cache,
+        EnergyBalanceInfo? result)
     {
-        // TODO: cache this somehow to not need to create a bunch of these which contain dictionaries to contain
-        // further items
-        var result = new EnergyBalanceInfo();
-
-        if (calculateRequiredResources)
+        if (result == null)
         {
-            result.SetupTrackingForRequiredCompounds();
+            result = new EnergyBalanceInfo();
+
+            if (calculateRequiredResources)
```

Comment:
I think it would be a lot clearer to remove the `calculateRequiredResources` parameter from this method as the parameter *only* works if the result passed in is null, so that's a very tricky and error-prone usage of this method.